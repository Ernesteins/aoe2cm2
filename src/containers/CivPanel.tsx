import {ApplicationState} from '../types';
import {connect} from 'react-redux';
import CivPanel from "../components/draft/CivPanel";
import ActionType, {actionTypeFromAction} from "../constants/ActionType";
import PlayerEvent from "../models/PlayerEvent";
import {Dispatch} from "redux";
import * as actions from "../actions";


export function mapStateToProps(state: ApplicationState) {
    let triggerAction: ActionType = ActionType.NOTHING;
    if (state.draft.preset && state.ownProperties.nextAction < state.draft.preset.turns.length) {
        triggerAction = actionTypeFromAction(state.draft.preset.turns[state.ownProperties.nextAction].action);
    }
    return {
        triggerAction,
        whoAmI: state.ownProperties.whoAmI,
        nextAction: state.ownProperties.nextAction,
        draft: state.draft,
        iconStyle: state.iconStyle.iconStyle,
    };
}

export function mapDispatchToProps(dispatch: Dispatch<actions.Action>) {
    return {
        onClickCivilisation: (playerEvent: PlayerEvent, callback: any) => dispatch(actions.clickOnCiv(playerEvent, callback))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(CivPanel);
