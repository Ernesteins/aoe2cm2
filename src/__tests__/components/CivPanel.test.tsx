import {shallow} from "enzyme";
import * as React from 'react';
import CivPanel from "../../components/draft/CivPanel";
import Civilisation from "../../models/Civilisation";
import CivPanelType from "../../constants/CivPanelType";

it('PICK CivPanels render correctly', () => {
    Civilisation.ALL.forEach((civ: Civilisation) => {
            renderCivPanelAndCheckSnapshot(CivPanelType.PICK, civ);
        }
    );
});

it('empty PICK CivPanel renders correctly', () => {
    renderCivPanelAndCheckSnapshot(CivPanelType.PICK);
});

it('BAN CivPanels render correctly', () => {
    Civilisation.ALL.forEach((civ: Civilisation) => {
            renderCivPanelAndCheckSnapshot(CivPanelType.BAN, civ);
        }
    );
});

it('empty BAN CivPanel renders correctly', () => {
    renderCivPanelAndCheckSnapshot(CivPanelType.BAN);
});

function renderCivPanelAndCheckSnapshot(civPanelType: CivPanelType, civ?: Civilisation) {
    const civpanel = shallow(<CivPanel civilisation={civ} active={false} civPanelType={civPanelType}/>);
    expect(civpanel).toMatchSnapshot();
}
