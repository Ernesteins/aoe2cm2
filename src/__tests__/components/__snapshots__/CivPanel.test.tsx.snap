// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BAN CivPanels render correctly 1`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Aztecs",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 2`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Berbers",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 3`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Britons",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 4`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Burmese",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 5`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Byzantines",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 6`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Celts",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 7`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Chinese",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 8`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Ethiopians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 9`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Franks",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 10`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Goths",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 11`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Huns",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 12`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Incas",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 13`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Indians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 14`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Italians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 15`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Japanese",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 16`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Khmer",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 17`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Koreans",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 18`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Magyars",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 19`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Malay",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 20`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Malians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 21`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Mayans",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 22`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Mongols",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 23`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Persians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 24`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Portuguese",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 25`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Saracens",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 26`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Slavs",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 27`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Spanish",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 28`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Teutons",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 29`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Turks",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 30`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Vietnamese",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 31`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Vikings",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 32`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Bulgarians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 33`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Cumans",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 34`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Lithuanians",
    }
  }
  t={[Function]}
/>
`;

exports[`BAN CivPanels render correctly 35`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Tatars",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 1`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Aztecs",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 2`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Berbers",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 3`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Britons",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 4`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Burmese",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 5`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Byzantines",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 6`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Celts",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 7`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Chinese",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 8`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Ethiopians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 9`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Franks",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 10`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Goths",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 11`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Huns",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 12`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Incas",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 13`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Indians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 14`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Italians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 15`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Japanese",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 16`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Khmer",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 17`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Koreans",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 18`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Magyars",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 19`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Malay",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 20`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Malians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 21`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Mayans",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 22`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Mongols",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 23`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Persians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 24`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 4,
      "isRandomlyChosenCiv": false,
      "name": "Portuguese",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 25`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Saracens",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 26`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 3,
      "isRandomlyChosenCiv": false,
      "name": "Slavs",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 27`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 2,
      "isRandomlyChosenCiv": false,
      "name": "Spanish",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 28`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Teutons",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 29`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Turks",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 30`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 5,
      "isRandomlyChosenCiv": false,
      "name": "Vietnamese",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 31`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 1,
      "isRandomlyChosenCiv": false,
      "name": "Vikings",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 32`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Bulgarians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 33`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Cumans",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 34`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Lithuanians",
    }
  }
  t={[Function]}
/>
`;

exports[`PICK CivPanels render correctly 35`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  civilisation={
    Civilisation {
      "gameVersion": 6,
      "isRandomlyChosenCiv": false,
      "name": "Tatars",
    }
  }
  t={[Function]}
/>
`;

exports[`empty BAN CivPanel renders correctly 1`] = `
<CivPanel
  active={false}
  civPanelType="ban"
  t={[Function]}
/>
`;

exports[`empty PICK CivPanel renders correctly 1`] = `
<CivPanel
  active={false}
  civPanelType="pick"
  t={[Function]}
/>
`;
